cuda/test/matrix/dense_kernels.cpp

```diff
@@ -190,6 +191,23 @@ class Dense : public ::testing::Test {
 };
 
 
+TEST_F(Dense, CudaCopyRespectsStride)
+{
+    set_up_vector_data(3);
+    auto result = Mtx::create(cuda, dx->get_size(), dx->get_size()[1] + 1);
+    double val = 123456789.0;
+    auto original_data = result->get_values();
+    auto padding_ptr = original_data + dx->get_size()[1];
+    cuda->copy_from(ref.get(), 1, &val, padding_ptr);
+
+    dx->convert_to(result.get());
+
+    GKO_ASSERT_MTX_NEAR(result, dx, 0);
```

Comment:
Does this actually compare the strides of the two matrices properly ?

Reply:
No, the comparison is based on `matrix_data`, which ignores strides entirely. But you do raise a point, in that I should check the stride remains unmodified.